frontend/src/app/shared/components/tree/tree.component.ts

```diff
@@ -0,0 +1,148 @@
+import { SelectionModel } from '@angular/cdk/collections';
+import { FlatTreeControl } from '@angular/cdk/tree';
+import { CommonModule } from '@angular/common';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatTreeFlatDataSource, MatTreeFlattener, MatTreeModule } from '@angular/material/tree';
+
+import { ChecklistDatabase } from './auxiliary/ChecklistDatabase';
+import { TreeDataObject } from './auxiliary/TreeDataObject';
+import { TreeItemFlatNode } from './auxiliary/TreeItemFlatNode';
+import { TreeItemNode } from './auxiliary/TreeItemNode';
+
+@Component({
+    selector: 'app-tree',
+    templateUrl: 'tree.component.html',
+    styleUrls: ['tree.component.sass'],
+    providers: [ChecklistDatabase],
+    standalone: true,
+    imports: [
+        MatTreeModule,
+        MatButtonModule,
+        MatCheckboxModule,
+        MatFormFieldModule,
+        MatInputModule,
+        MatIconModule,
+        CommonModule,
+    ],
+})
+export class TreeComponent implements OnInit {
+    @Input() asCheckList: boolean = false;
+
+    @Input() treeData: TreeDataObject[];
+
+    @Input() minHeightValue: number = 25;
+
+    @Output() selectionChanged = new EventEmitter<string[]>();
+
+    flatNodeMap = new Map<TreeItemFlatNode, TreeItemNode>();
+
+    /** Map from nested node to flattened node. This helps us to keep the same object for selection */
+    nestedNodeMap = new Map<TreeItemNode, TreeItemFlatNode>();
+
+    treeControl: FlatTreeControl<TreeItemFlatNode>;
+
+    treeFlattener: MatTreeFlattener<TreeItemNode, TreeItemFlatNode>;
+
+    dataSource: MatTreeFlatDataSource<TreeItemNode, TreeItemFlatNode>;
```

Comment:
Can't we use less material structures for this? 
As most likely we will have 1-2 level tree (type name -> item name)

+ it will be quite difficult to make any changes in your logic...

Reply:
@Anatolii-Ivanov so should I redo this without using material structures?